docs/build.sh

```diff
@@ -18,38 +18,74 @@
 # specific language governing permissions and limitations
 # under the License.
 
-set -e
+set -euo pipefail
 
-FWDIR="$(cd "`dirname "$0"`"; pwd)"
-cd "$FWDIR"
+MY_DIR="$(cd "`dirname "$0"`"; pwd)"
```

Comment:
```suggestion
MY_DIR="$(cd -p "$(dirname "$0")"; pwd)"
```

Reply:
Ah yeah. I would like to keep it to a separate PR. I already fixed a lot of SHELL problems by running shellcheck (part of the auto-commit changes). I will send a bunch (maybe 40) of other small PRs including pylint fixes and shell fixes, and xml fixes and tab removals etc - but I think they should be separate from that one.